File: Vanilla/sources/zdoc/media/lua/shared/Library/RadioAPI.lua
---@class RadioAPI : zombie.radio.RadioAPI
---@field private instance RadioAPI
RadioAPI = {}

---@public
---@param arg0 int
---@return int
function RadioAPI:timeStampToHours(arg0) end

---@public
---@param arg0 int
---@return int
function RadioAPI:timeStampToMinutes(arg0) end

---@public
---@param arg0 int
---@return int
function RadioAPI:timeStampToDays(arg0) end

---@public
---@return RadioAPI
function RadioAPI:getInstance() end

---@public
---@param arg0 String
---@return KahluaTable
function RadioAPI:getChannels(arg0) end

---@public
---@return boolean
function RadioAPI:hasInstance() end

---@public
---@param arg0 int
---@param arg1 int
---@param arg2 int
---@return int
function RadioAPI:timeToTimeStamp(arg0, arg1, arg2) end

File: Vanilla/sources/zomboid/zombie/radio/RadioAPI.java
package zombie.radio;

import java.util.Iterator;
import java.util.Map;
import java.util.Map.Entry;
import se.krka.kahlua.vm.KahluaTable;
import zombie.Lua.LuaManager;

public final class RadioAPI {
   private static RadioAPI instance;

   public static int timeToTimeStamp(int var0, int var1, int var2) {
      return var0 * 24 + var1 * 60 + var2;
   }

   public static int timeStampToDays(int var0) {
      return var0 / 1440;
   }

   public static int timeStampToHours(int var0) {
      return var0 / 60 % 24;
   }

   public static int timeStampToMinutes(int var0) {
      return var0 % 60;
   }

   public static boolean hasInstance() {
      return instance != null;
   }

   public static RadioAPI getInstance() {
      if (instance == null) {
         instance = new RadioAPI();
      }

      return instance;
   }

   private RadioAPI() {
   }

   public KahluaTable getChannels(String var1) {
      Map var2 = ZomboidRadio.getInstance().GetChannelList(var1);
      KahluaTable var3 = LuaManager.platform.newTable();
      if (var2 != null) {
         Iterator var4 = var2.entrySet().iterator();

         while(var4.hasNext()) {
            Entry var5 = (Entry)var4.next();
            var3.rawset(var5.getKey(), var5.getValue());
         }
      }

      return var3;
   }
}
